distributed/shuffle/_merge.py

```diff
@@ -0,0 +1,439 @@
+# mypy: ignore-errors
```

Comment:
Is this necessary? If so, let's add a follow-up ticket to clean this up.

Reply:
There will be a follow up task to clean up this layer after the pack/unpack protocol is dropped. I didn't want to go through this exercise for almost-dead-code.
Refactoring the layer makes sense as a follow up ticket but I don't think we should have a ticket just for mypy.